src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1810,20 +1810,28 @@ transaction BlockchainLMDB::get_tx(const crypto::hash& h) const
     get_result = mdb_cursor_get(m_cur_txs, &val_tx_id, &result, MDB_SET);
   }
   if (get_result == MDB_NOTFOUND)
-    throw1(TX_DNE(std::string("tx with hash ").append(epee::string_tools::pod_to_hex(h)).append(" not found in db").c_str()));
+    return false;
   else if (get_result)
     throw0(DB_ERROR(lmdb_error("DB error attempting to fetch tx from hash", get_result).c_str()));
 
   blobdata bd;
   bd.assign(reinterpret_cast<char*>(result.mv_data), result.mv_size);
 
-  transaction tx;
   if (!parse_and_validate_tx_from_blob(bd, tx))
     throw0(DB_ERROR("Failed to parse tx from blob retrieved from the db"));
 
   TXN_POSTFIX_RDONLY();
 
-  return tx;
+  return true;
+}
+
+transaction BlockchainLMDB::get_tx(const crypto::hash& h) const
+{
+  transaction tx;
+
+  if (!get_tx(h, tx))
+    throw1(TX_DNE(std::string("tx with hash ").append(epee::string_tools::pod_to_hex(h)).append(" not found in db").c_str()));
+  return std::move(tx);
```

Comment:
I would [drop the `std::move`](http://stackoverflow.com/a/17473869). There are some people in those comments that disagree for some reason. Basically the compiler tries to construct on the callers stack, then tries to do a move, then tries to do a copy.

There is also returning by `boost::optional<>`. It has internally unitialized memory for the object, so it is pretty efficient, especially with RVO/moves.